include/deal.II/dofs/dof_tools.h

```diff
@@ -1887,10 +1919,26 @@ namespace DoFTools
    * degrees of freedom on these. In other words, this function can only work
    * if all cells in the patch are either locally owned or ghost cells.
    */
-  template <typename DoFHandlerType>
+  template <int dim, int spacedim>
   std::vector<types::global_dof_index>
   get_dofs_on_patch(
-    const std::vector<typename DoFHandlerType::active_cell_iterator> &patch);
+    const std::vector<typename DoFHandler<dim, spacedim>::active_cell_iterator>
+      &patch);
+
+  /**
+   * The same as above.
+   *
+   * @deprecated Use the function that takes dim and spacedim as template
+   *   argument in favor of the following function.
```

Comment:
```suggestion
   *   argument.
```
(technically, that other function comes before).